services/collaboration/pkg/middleware/wopicontext.go

```diff
@@ -230,6 +245,13 @@ func GenerateWopiToken(wopiContext WopiContext, cfg *config.Config, st microstor
 	return accessToken, claims.ExpiresAt.UnixMilli(), err
 }
 
+func (w *WopiContext) GetPublicShare() *link.PublicShare {
```

Comment:
This is questionable as it is at the moment. Why the `publicShare` property isn't public like the rest? That would make this method obsolete. At the moment, this method doesn't add anything of value other than getting the value, so making the property public to keep the consistency with the rest of the data seems fine to me.

Reply:
So the property needs to be private in order not to include it in the jwt token, and we need a public method to access to the private property.

A couple of additional things:
* If the method is attached to the WopiContext, it's better to have the code as close as possible to the WopiContext definition, so the code should be move upwards. Reading the code should be easier this way.
* This method just returns the private property, but it might not be set or evaluated yet (a bit weird scenario, but it could happen if we prepare a WopiContext outside of the http middleware). In this regard, it might be better to either include a public `EvaluatePublicShare` method that should be called before getting the public share (most of the code is already in the middleware, so it's moving things around), or extract the public share information on demand with the `GetPublicShare` method and let the private property act as a cache.
  I'd go with the `EvaluatePublicShare` in case the access token changes (which shouldn't happen under normal conditions) because it gives us extra flexibility. As long as it's documented that the `EvaluatePublicShare` must be called before the `GetPublicShare`, it should be fine.

Reply:
I don't think that `EvaluatePublicShare` method will be good because it has some dependencies.
We could change the private field `publicShare` to `scope` and change the `GetPublicShare` to generic `GetScope(keyPrefix string, m proto.Message)` method

Reply:
The scope is only filled in the middleware, and I don't think it's a good idea. If you check the tests, we create a custom WopiContext, but even if we supply correct data to the public WopiContext properties, the scope won't return anything valuable. This is why a suggested to include the `EvaluatePublicShare` method: to fill and adjust the private properties.
Changing the WopiContext, in particular the access token inside, could also have negative consequences because the scope information isn't valid any longer and we can't update the information in any way. This means that the only valid context is the one supplied by the middleware, and changing or adjusting data in the WopiContext is discouraged.

Reply:
We can try to do it another way, use the context like for the user?
```
ctx = ctxpkg.ContextSetScopes(ctx, scope)
```
https://github.com/owncloud/ocis/blob/2237534fc3e3ee97e7ea1b7a5eab0eab8005cf02/services/collaboration/pkg/middleware/wopicontext.go#L123-L134

In this case we can use existing functions `ContextSetScopes`, `ContextGetScopes` and add some scope extractor function